core/metamodel/src/main/java/org/apache/isis/core/metamodel/facets/collections/javautilcollection/JavaArrayFacet.java

```diff
@@ -67,29 +73,32 @@ public void init(final ObjectAdapter collectionAdapter, final ObjectAdapter[] in
      * Expected to be called with a {@link ObjectAdapter} wrapping an array.
      */
     @Override
-    public ObjectAdapter firstElement(final ObjectAdapter collectionAdapter) {
-        final Object[] array = array(collectionAdapter);
-        return array.length > 0 ? getAdapterManager().getAdapterFor(array[0]) : null;
+    public ObjectAdapter firstElement(final ObjectAdapter arrayAdapter) {
+        final Object[] array = pojoArray(arrayAdapter);
+        if(isEmpty(array)) {
+            return null;
+        }
+        return array.length > 0 ? getObjectAdapterProvider().adapterFor(array[0]) : null;
```

Comment:
change in behavior:
BEFORE: only a map lookup, which would not trigger ObjectAdapter creation
AFTER: triggers ObjectAdapter creation if required

Reply:
I think this is ok.